### PR TITLE
[Core][WIP] Make the object in TorrentBroadcast as transient

### PR DESCRIPTION
**Pull request by viirya**
## What changes were proposed in this pull request?

WIP

## How was this patch tested?

Jenkins test.

Please review http://spark.apache.org/contributing.html before opening a pull request.

### CODE REVIEW

**Comment by viirya** [comment]
I am not sure if I am missing something. But looks like we serialize the object in `TorrentBroadcast`? If I understand it wrongly, I'd close this.

cc @cloud-fan @hvanhovell

File: core/src/main/scala/org/apache/spark/broadcast/TorrentBroadcast.scala [hunk 1]
```diff
@@ -54,7 +54,7 @@ import org.apache.spark.util.io.{ChunkedByteBuffer, ChunkedByteBufferOutputStrea
  * @param obj object to broadcast
  * @param id A unique identifier for the broadcast variable.
  */
-private[spark] class TorrentBroadcast[T: ClassTag](obj: T, id: Long)
+private[spark] class TorrentBroadcast[T: ClassTag](@transient val obj: T, id: Long)
```
Comment (scwf) [inline core/src/main/scala/org/apache/spark/broadcast/TorrentBroadcast.scala]: For the old code, obj is not a field, it is just a ctor argument

Comment (viirya) [inline core/src/main/scala/org/apache/spark/broadcast/TorrentBroadcast.scala]: I think the compiler will automatically create a field for it?

Comment (viirya) [inline core/src/main/scala/org/apache/spark/broadcast/TorrentBroadcast.scala]: OK. I see. As there is no other places except `val numBlocks: Int = writeBlocks(obj)` use it, there is no field created for it.

**Pull request closed without merging**